src/ResourceManager/IotHub/Commands.IotHub/IotHub/SetAzureRmIotHub.cs

```diff
@@ -122,6 +126,27 @@ public class SetAzureRmIotHub : IotHubBaseCmdlet
         [ValidateNotNullOrEmpty]
         public PSOperationsMonitoringProperties OperationsMonitoringProperties { get; set; }
 
+        [Parameter(
+            ParameterSetName = UpdateRoutingPropertiesParameterSet,
+            Mandatory = false,
+            HelpMessage = "RoutingProperties")]
+        [ValidateNotNullOrEmpty]
+        public PSRoutingProperties RoutingProperties { get; set; }
+
+        [Parameter(
+            ParameterSetName = UpdateRoutePropertiesParameterSet,
+            Mandatory = false,
+            HelpMessage = "Routes")]
+        [ValidateNotNullOrEmpty]
+        public List<PSRouteProperties> Routes { get; set; }
+
+        [Parameter(
+            ParameterSetName = UpdateFallbackRoutePropertyParameterSet,
+            Mandatory = false,
+            HelpMessage = "Fallback Routes")]
+        [ValidateNotNullOrEmpty]
+        public PSFallbackRouteProperties FallbackRoute { get; set; }
```

Comment:
We should rename the rout property types to make the distinction clear, the List<> type should be an array, and parameter names should be singular.